File: Offer2/0045/main.go
package main

import (
	"strconv"
	"strings"
)

func minNumber(nums []int) string {
	array := []string{}
	for _, num := range nums {
		array = append(array, strconv.Itoa(num))
	}
	length := len(array)
	for i := 0; i < length-1; i++ {
		for j := i + 1; j < length; j++ {
			if array[i]+array[j] > array[j]+array[i] {
				array[i], array[j] = array[j], array[i]
			}
		}
	}
	return strings.Join(array, "")
}

File: Offer2/0045/main.py
from typing import List


class Solution:
    def minNumber(self, nums: List[int]) -> str:
        array: List[str] = [str(x) for x in nums]
        for i in range(len(array) - 1):
            for j in range(i + 1, len(array)):
                if array[i] + array[j] > array[j] + array[i]:
                    array[i], array[j] = array[j], array[i]
        return "".join(array)
